appendix-memorymap.tex

```diff
@@ -367,34 +367,30 @@ \section{CPU Memory Banking}
+then the other banking mechanisms do not apply. This is true even if the
+offset is 0, allowing the 16-bit addresses to access RAM in bank 0 (such as
+address 0.D000).
+
+C65-style \$D030 banking and C64-style \$00 / \$01 banking both select 8KB
+regions to map to bank 2, which (by default) contains C64 ROM code. These two
+mechanisms overlap in which regions they can map to ROM. If either mechanism
+maps a region to ROM (and it is not mapped elsewhere by the MAP register),
+then it is mapped to ROM.
 
 The following diagram shows the different types of banking that can
-apply to the different areas of the 64KB that the CPU can see.  The
-higher layers take priority over the lower layers, as described in the
-previous paragraph.
+apply to the different areas of the 64KB that the CPU can see.
+
 \begin{center}
 \begin{tabular}{rccccccc}
 \cline{4-5} \cline{7-8}
+\rowcolor[HTML]{34FF34}
+\multicolumn{1}{r|}{\cellcolor[HTML]{C0C0C0}MAP}    & \multicolumn{2}{c|}{\cellcolor[HTML]{34FF34}\begin{tabular}[c]{@{}c@{}}MAP LO\\ (4 x 8KB slabs)\end{tabular}} & \multicolumn{5}{c|}{\cellcolor[HTML]{34FF34}\begin{tabular}[c]{@{}c@{}}MAP HI\\ (4 x 8KB slabs)\end{tabular}}                                                                                                                                                                                                                                                                                                                                                                                                     \\ \cline{2-8}
 \rowcolor[HTML]{C0C0C0} I/O/CART &                                 & \multicolumn{1}{c|}{\cellcolor[HTML]{C0C0C0}}                               & \multicolumn{1}{c|}{\cellcolor[HTML]{FE0000}\begin{tabular}[c]{@{}c@{}}CART\\ ROMLO\end{tabular}} & \multicolumn{1}{c|}{\cellcolor[HTML]{FE0000}\begin{tabular}[c]{@{}c@{}}CART\\ ROMHI\end{tabular}} & \multicolumn{1}{c|}{\cellcolor[HTML]{C0C0C0}}                                                      & \multicolumn{1}{c|}{\cellcolor[HTML]{F8A102}I/O}                                                 & \multicolumn{1}{c|}{\cellcolor[HTML]{FE0000}\begin{tabular}[c]{@{}c@{}}CART \\ ROMHI\end{tabular}} \\ \cline{4-8}
 \rowcolor[HTML]{EFEFEF}
 C65                                                 &                                 & \multicolumn{1}{c|}{\cellcolor[HTML]{EFEFEF}}                               & \multicolumn{1}{c|}{\cellcolor[HTML]{F8FF00}BASIC}                                                & \multicolumn{1}{c|}{\cellcolor[HTML]{F8FF00}BASIC}                                                & \multicolumn{1}{c|}{\cellcolor[HTML]{F8FF00}\begin{tabular}[c]{@{}c@{}}INTER-\\ FACE\end{tabular}} & \multicolumn{1}{c|}{\cellcolor[HTML]{EFEFEF}}                                                   & \multicolumn{1}{c|}{\cellcolor[HTML]{F8FF00}KERNAL}                                                \\ \cline{2-8}
```

Comment:
Maybe we call this line $D030 instead of C65 as it is not banking any C65 specific ROM parts.
Also, the section mapped at $8000 will come from $2.8000, which is not BASIC (neither C65 nor C64 BASIC), is it? I believe the ROM at $2.8000 was reserved and yet unused.

Reply:
Accepted all changes. I wonder if we'll delete this diagram in a future revision that makes this chapter more concise. Its primary purpose is to clarify the confusing resolution order, which is no longer an issue after this change. But it's also nice to see the banking-affected regions laid out, for completeness.